Packages/Sequence-Unity/Sequence/SequenceSDK/EmbeddedWallet/EOAWalletToEmbeddedWalletAdapter.cs

```diff
@@ -0,0 +1,252 @@
+                }
+                
+                if (Wallet is Sequence.Wallet.EOAWallet wallet)
+                {
+                    if (waitForReceipt)
+                    {
+                        try
+                        {
+                            var responseReceipt = await Wallet.SendTransactionAndWaitForReceipt(client, transaction);
+
+                            if (responseReceipt == null)
+                            {
+                                throw new Exception("No Receipt");
+                            }
+                            else
+                            {
+                                var successfullReturn = new SuccessfulTransactionReturn(responseReceipt.transactionHash, null, null, null, JObject.FromObject(responseReceipt));
+                                successfulTransactionReturns.Add(successfullReturn);
+                                OnSendTransactionComplete?.Invoke(successfullReturn);
+                            }
+                        }
+                        catch (Exception receiptEx)
+                        {
+                            Debug.LogError("Error while waiting for the receipt: " + receiptEx.Message);
+                            string receiptExString = receiptEx.Message;
+                            var failedReturn = new FailedTransactionReturn(receiptEx.Message, null, null);
+                            failedTransactionReturns.Add(failedReturn);
+                            OnSendTransactionFailed?.Invoke(failedReturn);
+                        }
+                    }
+                    else
+                    {
+                        var responseReceipt = await Wallet.SendTransaction(client, transaction);
+                        successfulTransactionReturns.Add(new SuccessfulTransactionReturn(responseReceipt, null, null, null));
+                    }
+                }
+            }
+
+            if (failedTransactionReturns.Count == 0)
+            {
+                return successfulTransactionReturns.Count > 1
+                    ? new SuccessfulBatchTransactionReturn(successfulTransactionReturns.ToArray())
+                    : successfulTransactionReturns[0];
+            }
+
+            return failedTransactionReturns.Count > 1
+                ? new FailedBatchTransactionReturn(successfulTransactionReturns.ToArray(), failedTransactionReturns.ToArray())
+                : failedTransactionReturns[0];
+        }
+        
+        public event Action<SuccessfulContractDeploymentReturn> OnDeployContractComplete;
+        public event Action<FailedContractDeploymentReturn> OnDeployContractFailed;
+
+        public async Task<ContractDeploymentReturn> DeployContract(Chain network, string bytecode, string value)
+        {
+            var client = new SequenceEthClient(network);
+
+
+            ContractDeploymentReturn creturn;
```

Comment:
nit: unless you're using a common short form or acronym, we should lean towards verbose variable names as it makes it easier for the reader to know what they do
Consider renaming to deploymentReturn or contractDeploymentReturn